### PR TITLE
fix(dashboard): dashboard doesn't load properly if it has tabs

### LINKED ISSUE #20748: Dashboard shows endless loading
Upon initially loading a dashboard filters load and populate while the dashboard spins it's logo endlessly. A manual refresh causes a normal loading behavior. 

#### How to reproduce the bug
See screenshot. 
1. Go to any dashboard 
2. Wait for it to load
3. Hit refresh on your browser to get it to load correctly after waiting. 

### Expected results

Dashboard fully loads. 

### Actual results

Dashboard only loads after refresh. 

#### Screenshots

https://user-images.githubusercontent.com/1770898/179555764-5a31249f-144a-479a-bdf4-b4dbb4f74938.mp4

### Environment

- Windows 10,11 and Firefox, Chrome, Safari, and Edge. 
- Superset Version 1.3, 1.5, 2.0 
- docker-compose install currently version 2.0 but tried on all of the recent versions. 
- python 3.8.10

- [x ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

This has been an annoying issue that prevents us from rolling this out to customers.

### LINKED ISSUE #20797: Dashboard does not load when a filter is configured "Select first filter value by default" option
It seems when "Select first filter value by default" is configured for a filter and the dashboard has tabs the dashboard stays in a loading state and filters remain out of scope. Clicking the filters that are out of scope triggers the fetch for the first value and the dashboard can render. 

#### How to reproduce the bug

- Import the attached dashboard export (you might have to edit the database to use your examples db instead of the SQLLite one I used). 
- load the imported "video game sales" dashboard. 
- You should see the bug that looks like the attached screenshot
- click on the "filters out of scope" and the dashboard should load correctly. Refreshing the dashboard (as long as the url contains the `native_filters_key` query param, also results in the dashboard loading correctly.

### Expected results

The dashboard loads correctly when opened from the dashboard list/ home page or a link (that does not contain the `native_filters_key` query param. 

### Actual results

The dashboard stays in a loading state until "filters out of scope" is clicked or the dashboard is refreshed

#### Screenshots

Screen Shot 2022-07-20 at 1 52 56 PM

### Environment

 version `1.5.1`

### Additional context

Dashboard export:
[dashboard_export_20220720T134744.zip](https://github.com/apache/superset/files/9154399/dashboard_export_20220720T134744.zip)

### LINKED ISSUE #20930: [Dashboard] Issue with tabbed dashboard
There are 2 issues:
1. When user leaves dashboard on any tab other than the 1st tab, that tab is seen selected when user returns to dashboard. Even though the charts from 1st tab are visible.
2. In first situation, if user clicks on 1st tab and then again to previous tab, the chart data loading gets stuck.

#### How to reproduce the bug

1. Open Sample Dashboard, 'Sales Dashboard'.
2. Go to 'Exploratory' tab
3. Go to 'Dashboards' list page
4. Again, open 'Sales Dashboard'
5. 1st issue
6. Click on 'Sales Overview' tab
7. Click on 'Exploratory' tab
8. 2nd issue

### Expected results

1st issue: Dashboard opens with charts from 1st tab, and 1st tab seen selected.
2nd issue: Dashboard should load the charts normally.

### Actual results

1st issue: Dashboard open with charts from 1st tab, but 2nd tab is seen as selected.
2nd issue: Dashboard cannot load data in the charts.

#### Screenshots
dashboard_tab

### Environment

(please complete the following information):

- browser type and version: Chrome
- superset version: `2.0.0`
- Docker image
- any feature flags active: DASHBOARD_NATIVE_FILTERS

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

It is possible that both issues are related, solving 1st may resolve 2nd also.

### PR DESCRIPTION
**Pull request by stephenLYZ**
### SUMMARY

This PR fixes the problem that dashboard does not load properly after switching dashboard when there are tabs and native filters. The reason is that the dashboard is loading the tab component with incorrect timing.

#### before

https://user-images.githubusercontent.com/11830681/192158153-15686b05-60de-4a32-bb80-79e86ab87fea.mov

#### after

https://user-images.githubusercontent.com/11830681/192158119-ab5d5fcd-c1ae-4efc-b950-21d2bbdc4d49.mov

### ADDITIONAL INFORMATION

- [x] Has associated issue: 
- fixes https://github.com/apache/superset/issues/20797
- fixes https://github.com/apache/superset/issues/20748
- fixes https://github.com/apache/superset/issues/20930
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/gridComponents/Tabs.test.jsx [hunk 1]
```diff
@@ -202,4 +203,18 @@ describe('Tabs', () => {
     expect(modalMock.mock.calls).toHaveLength(1);
     expect(deleteComponent.callCount).toBe(0);
   });
+
+  if (
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Tabs.test.jsx:207]:
```suggestion
  it (
```

File: superset-frontend/src/dashboard/components/gridComponents/Tabs.test.jsx [hunk 2]
```diff
@@ -202,4 +203,18 @@ describe('Tabs', () => {
...
+      const wrapper = shallow(<Tabs {...props} />);
+      expect(wrapper.state('activeKey')).toBe('TABS_ID');
+      wrapper.setProps({
+        ...props,
+        dashboardId: 2,
+        component: dashboardLayoutWithTabs.present.TAB_ID,
+      });
+      expect(wrapper.state('activeKey')).toBe('TAB_ID');
+    })
+  );
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Tabs.test.jsx:219]:
```suggestion
  it('should set new tab key if dashboardId was changed', () => {
    const wrapper = shallow(<Tabs {...props} />);
    expect(wrapper.state('activeKey')).toBe('TABS_ID');
    wrapper.setProps({
      ...props,
      dashboardId: 2,
      component: dashboardLayoutWithTabs.present.TAB_ID,
    });
    expect(wrapper.state('activeKey')).toBe('TAB_ID');
  });
```

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Comment by kamalkeshavani-aiinside** [comment]
@stephenLYZ I tried 2.0.1rc2 docker hub image, to test this fix. I can still reproduce some issue of tabs. Please check.
dashboard_tab_issue

**Comment by sfirke** [comment]
I can confirm that the issue of a dashboard's first (default) tab not loading if there are filters out of scope persists using the 2.0.1rc2 image on docker hub.

**Comment by sfirke** [comment]
@duanzy47 this will not be fixed until 2.1.0, see this comment for more https://github.com/apache/superset/discussions/21263#discussioncomment-4180734

**Comment by sfirke** [comment]
Looks like the issue persists in 2.1.0rc1.

**Pull request merged**